src/hoc/ForceDisableServerRender.tsx

```diff
@@ -0,0 +1,17 @@
+'use client'
+
+import { useState, useEffect, ReactNode } from 'react'
+
+//this HOC can be used to force disable server rendering for the components supposed to render only on the client side
+export const ForceDisableServerRender = ({ children }: { children: ReactNode }) => {
+  //this state tracks if the component is in client side or not during hydration
+  const [isCLient, setIsClient] = useState(false)
+
+  useEffect(() => {
+    setIsClient(true)
+  }, [])
+
+  if (!isCLient) return null
+
+  return children
```

Comment:
I'm sure you're doing this for a valid reason.. but what is the reason behind checking `isClient` for a client component? Aren't client components only supposed to render on the client side after hydration? So by that logic what does `if (!isClient) return null` do?

Reply:
In NextJS, any component that is placed in the server component (such as page.tsx) are server rendered even though the component has `use client` directive. The `use client` directive simply means the component is a client component and thus we are allowed to use client side states, hooks, and effects in such a components. 

In the above case, the `ForceDisableServerRender` is used in the server component (page.tsx). Initially, NextJS renders the page.tsx server component in the server and thus this component also runs and renders in the server itself. However, initially in the server, since this is a client component, the `isClient` state would be undefined and this component would return `null` in the server rendering nothing at all. This approach is used to get rid of hydration errors. If we don't return `null` in the server then the server would still return `children` and pre-render the `children`, and then when the pre-rendered html tries to hydrate with the client side html there will be a mismatch scenario which would create the hydration error. Thus, this component would force stop pre-rendering in the server side to avoid hydration error.

Reply:
Yeah client components are also rendered on the server, while the server components are rendered and run exclusively on the server.

Reply:
Since we are returning `null` instead of our component until hydration, wouldn't that degrade performance instead?

Reply:
There is no issue with performance because 
1. Even though we don't return `null`, the server would return the `children` and the hydration would still take place and would look like  ***server returned `children` X client side `children`***
2. If we return `null` in the server then the server would return `null` to the client and during hydration it would look like ***server returned `null` X client side `children`***

where X means "hydrates with". 

Client side `children` is returned in the client side after states and effects are run while server returned `children` is returned on the server side without running states and effects.  

cc @priosshrsth

Reply:
This is not an ideal solution. But I don't think it should have noticable performance impact. For the record @aatbip I think we can ignore the hyrdration error if we want. That won't show up in production. 

But this seems ok as well, because we are always showing empty message anyway, and the data are not empty in reality. So shouldn't have impact on seo as well. I assume the UI has some sort of loader though. Otherwise, like @rrojan mentioned, user would be flashed with blank UI which switcher to content all of a sudden.